### PR TITLE
[CELEBORN-1609] Support SSL for celeborn RESTful service

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?
Support SSL for celeborn RESTful service.

### Why are the changes needed?
For HTTP SSL connection requirements.

### Does this PR introduce _any_ user-facing change?
No, SSL is disabled by defaults.

### How was this patch tested?

Integration testing.

```
celeborn.master.http.ssl.enabled=true
celeborn.master.http.ssl.keystore.path=/hadoop/keystore.jks
celeborn.master.http.ssl.keystore.password=xxxxxxx
```
image

image

### CODE REVIEW

**Comment by turboFei** [comment]
cc @mridulm @pan3793 @RexXiong @waitinfuture

File: service/src/main/scala/org/apache/celeborn/server/common/http/api/CelebornOpenApiResource.scala [hunk 1]
```diff
@@ -81,9 +82,10 @@ class CelebornOpenApiResource extends BaseOpenApiResource with ApiRequestContext
     }
   }
 
-  private def setCelebornOpenAPIDefinition(openApi: OpenAPI, requestBaseUrl: String): OpenAPI = {
-    // TODO: to improve when https is enabled.
-    val apiUrls = List(requestBaseUrl, s"http://${httpService.connectionUrl}/").distinct
+  private def setCelebornOpenAPIDefinition(openApi: OpenAPI, requestBaseUri: URI): OpenAPI = {
+    val httpScheme = if (httpService.httpSslEnabled()) "https:" else "http:"
+    val requestBaseUrl = s"$httpScheme${requestBaseUri.getSchemeSpecificPart}"
+    val apiUrls = List(requestBaseUrl, s"$httpScheme//${httpService.connectionUrl}/").distinct
```
Comment (turboFei) [inline service/src/main/scala/org/apache/celeborn/server/common/http/api/CelebornOpenApiResource.scala:88]: To fix the swagger request url scheme inconsistent when SSL enabled.
image

Comment (turboFei) [reply service/src/main/scala/org/apache/celeborn/server/common/http/api/CelebornOpenApiResource.scala:88]:
```
scala> val uri = new java.net.URI("https://celeborn.vip.com:9098/").getSchemeSpecificPart
val uri: String = //celeborn.vip.com:9098/

scala> 
```

**Comment by RexXiong** [comment]
Thanks. merge to main(v0.6.0)

**Pull request closed without merging**